specifyweb/frontend/js_src/lib/components/Molecules/AttachmentPicker.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import { preferencesText } from '../../localization/preferences';
+import { wbText } from '../../localization/workbench';
+import { Tabs } from '../AppResources/Tabs';
+import { Button } from '../Atoms/Button';
+import { Textarea } from '../Atoms/Form';
+import { AttachmentsView } from '../Attachments';
+import { fetchOriginalUrl } from '../Attachments/attachments';
+import { UploadAttachment, useAttachment } from '../Attachments/Plugin';
+import { LoadingContext } from '../Core/Contexts';
+import { serializeResource } from '../DataModel/helpers';
+import type { SerializedResource } from '../DataModel/helperTypes';
+import type { Attachment } from '../DataModel/types';
+import { Dialog } from './Dialog';
+
+const types = ['url', 'image', 'attachments', 'attachments'] as const;
+
+export function AttachmentPicker({
+  url,
+  onChange: handleChange,
+  isReadOnly,
+  displayScale,
+}: {
+  readonly url: string | undefined;
+  readonly onChange: (url: string) => void;
+  readonly isReadOnly: boolean;
+  readonly displayScale?: boolean;
+}): JSX.Element {
+  const [attachment, setAttachment] = useAttachment(undefined);
+  useErrorContext('attachment', attachment);
+
+  const [isOpen, , , handleToggle] = useBooleanState();
+
+  const loading = React.useContext(LoadingContext);
+
+  const [urlNotFound, setUrlNotFound] = React.useState(false);
+
+  const [type, setType] = React.useState<
+    'attachments' | 'attachments' | 'image' | 'url'
+  >('url');
+
+  function handleAttachment(attachment: SerializedResource<Attachment>): void {
+    loading(
+      fetchOriginalUrl(attachment).then((url) => {
+        url === undefined ? setUrlNotFound(true) : handleChange(url);
+      })
+    );
+    handleToggle();
+  }
+
+  return (
+    <>
+      {!isReadOnly && (
+        <Button.Gray onClick={() => handleToggle()}>
+          {url === undefined ? commonText.pickImage() : commonText.change()}
+        </Button.Gray>
+      )}
+
+      {url !== undefined && !isReadOnly ? (
+        <Button.Gray
+          onClick={() => {
+            setAttachment(undefined);
+            handleChange('default');
```

Comment:
sorry that I overlooked this earlier.
instead of `'default'` it should use `undefined` here - this component should not now anything about `'default'` as that's something specif to the welcome page pref